circuit/src/ecc/generic/ecdsa.rs

```diff
@@ -0,0 +1,219 @@
+use std::vec;
+
+use crate::{
+	ecc::generic::native::EcPoint,
+	integer::native::Integer,
+	params::{ecc::EccParams, rns::RnsParams},
+	FieldExt,
+};
+
+use halo2::{
+	arithmetic::Field,
+	halo2curves::{
+		ff::{FromUniformBytes, PrimeField},
+		group::Curve,
+		secp256k1::{Fp, Fq, Secp256k1, Secp256k1Affine},
+		CurveAffine,
+	},
+};
+use num_bigint::BigUint;
+use rand::thread_rng;
+
+/// Helper function to convert Fp element to Fq element
+pub fn fp_to_fq(x: Fp) -> Fq {
+	let mut x_repr = [0u8; 32];
+	x_repr.copy_from_slice(x.to_repr().as_ref());
+	let mut x_bytes = [0u8; 64];
+	x_bytes[..32].copy_from_slice(&x_repr[..]);
+	Fq::from_uniform_bytes(&x_bytes)
+}
+
+/// Helper function converts BigUint to Fq
+pub fn biguint_to_fq(x: BigUint) -> Fq {
+	let order_biguint = BigUint::from_bytes_be(Fq::MODULUS.as_bytes());
+	let x = num_integer::Integer::mod_floor(&x, &order_biguint);
+	let mut x_le = x.to_bytes_le();
+	let x_le_len = x_le.len();
+	let zeros_vec = vec![0u8; 64 - x_le_len];
+	x_le.extend(zeros_vec);
+	Fq::from_uniform_bytes(x_le[..].try_into().unwrap())
+}
```

Comment:
Can we use `big_to_fe` here, from utils?

Reply:
Done.